pkg/kapp/resources/identified_resources_list.go

```diff
@@ -25,6 +26,11 @@ func (r IdentifiedResources) List(labelSelector labels.Selector, resRefs []Resou
 		return nil, err
 	}
 
+	// avoid listing labeled resources for newly created app
+	if opts.NewApp {
+		return nil, nil
+	}
```

Comment:
what about call above that happens and gets throw away?

Reply:
In that call results are being memoized [here](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/resources/resource_types.go#L157) that is why added return after that.

Reply:
I guess the question then is, do we need that memoized list if it's a new app?

Reply:
hmm, nice question. Will have a look again to check in which cases this memoized info are being used.

Reply:
At this stage kapp is using memoized info, which is being memoized at [this](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/cmd/app/deploy.go#L145) stage while preparing newResources, so we can skip this and return bit early. Updated changes as well.